### PR TITLE
ENH: Support for Gemma 2 and Llama 3.1 Models for vllm & sglang

### PR DESCRIPTION
**Pull request by vikrantrathore**
This pull request includes the following updates:

1. **Model Support**:
   - Added support for the Gemma 2 and Llama 3.1 models.

2. **Package Updates**:
   - Upgraded `sglang`, `vllm`, and `Flashinfer` packages to their latest versions.

3. **New Feature**:
   - Implemented the `--cpu-offload-gb` feature in `vllm` 0.5.3, enabling CPU offloading for large context windows when the model cannot fit in GPU VRAM.

**Requirements**:
- CUDA version 12.1 or higher is necessary to support these changes.
- Python 3.11 is required for compatibility with the updated packages and features.

Please review the changes and ensure that your environment meets the necessary requirements before merging.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -140,8 +141,10 @@ transformers =
     peft
 vllm =
     vllm>=0.2.6
+    flashinfer @ https://github.com/flashinfer-ai/flashinfer/releases/download/v0.1.1/flashinfer-0.1.1+cu121torch2.3-cp311-cp311-linux_x86_64.whl  ; sys_platform=='linux'  #for gemma 2 which uses Fashinfer for inference
 sglang =
-    sglang[all]
+    sglang>=0.1.22 ; sys_platform=='linux'
+    flashinfer @ https://github.com/flashinfer-ai/flashinfer/releases/download/v0.1.1/flashinfer-0.1.1+cu121torch2.3-cp311-cp311-linux_x86_64.whl  ; sys_platform=='linux'  #for gemma 2 which uses Fashinfer for inference
```
Comment (ChengjieLi28) [inline setup.cfg:147]: You cannot limit the package version here. Why do users have to use `cp311` and `cu121`?

**Comment by vikrantrathore** [comment]
I have made the suggested changes except changing the flashinfer which requires a extensive build for removing the cuda and python requirements.

**Comment by qinxuye** [comment]
From the sglang doc and the setup related files, the flashinfer seems an option, how about adding the flashinfer into the document https://inference.readthedocs.io/en/latest/getting_started/installation.html#sglang-backend to tell user for cuda environment, install flashinfer first?

**Comment by vikrantrathore** [comment]
> From the sglang doc and the setup related files, the flashinfer seems an option, how about adding the flashinfer into the document https://inference.readthedocs.io/en/latest/getting_started/installation.html#sglang-backend to tell user for cuda environment, install flashinfer first?

Removed flashinfer and added it to documentation.

**Comment by qinxuye** [comment]
The lint failed, for Xinference, see https://github.com/xorbitsai/inference/actions/runs/10165496975/job/28113525447?pr=1929 to fix it.

File: xinference/model/llm/vllm/core.py [hunk 2]
```diff
@@ -151,11 +151,12 @@ class VLLMGenerateConfig(TypedDict, total=False):
     VLLM_SUPPORTED_CHAT_MODELS.append("c4ai-command-r-v01")
 
 if VLLM_INSTALLED and vllm.__version__ >= "0.5.3":
+    VLLM_SUPPORTED_CHAT_MODELS.append("gemma-2-it")
     VLLM_SUPPORTED_CHAT_MODELS.append("mistral-nemo-instruct")
     VLLM_SUPPORTED_CHAT_MODELS.append("mistral-large-instruct")
 
 if VLLM_INSTALLED and vllm.__version__ > "0.5.3":
-    VLLM_SUPPORTED_MODELS.append("llama-3.1")
+    VLLM_SUPPORTED_CHAT_MODELS.append("llama-3.1")
```
Comment (qinxuye) [inline xinference/model/llm/vllm/core.py:159]:
```suggestion
    VLLM_SUPPORTED_MODELS.append("llama-3.1")
```

**Review by qinxuye — APPROVED** [review]

**Pull request merged**